packages/web/components/funnels/concentrated-liquidity/use-cfmm-to-cl-migration.ts

```diff
@@ -0,0 +1,98 @@
+import { CoinPretty } from "@keplr-wallet/unit";
+import { useCallback, useMemo } from "react";
+
+import { useStore } from "~/stores";
+
+export type MigrationParams =
+  | { cfmmShares: CoinPretty }
+  | { lockIds: string[] };
+
+/** Use for sending CFMM to CL migration messages if applicable. */
+export function useCfmmToClMigration(cfmmPoolId: string): {
+  isLinked: boolean;
+  userCanMigrate: boolean;
+  linkedClPoolId: string | null;
+  migrate: (params: MigrationParams) => Promise<void>;
+} {
+  const {
+    queriesStore,
+    chainStore: {
+      osmosis: { chainId },
+    },
+    accountStore,
+    derivedDataStore,
+  } = useStore();
+
+  const osmosisAccount = accountStore.getAccount(chainId).osmosis;
+  const osmosisQueries = queriesStore.get(chainId).osmosis!;
+  const { sharePoolDetail } = derivedDataStore.getForPool(cfmmPoolId);
+
+  const userCanMigrate =
+    !sharePoolDetail.userAvailableShares.toDec().isZero() ||
+    sharePoolDetail.userLockedAssets.flatMap(({ lockIds }) => lockIds).length >
+      0 ||
+    sharePoolDetail.userUnlockingAssets.flatMap(({ lockIds }) => lockIds)
+      .length > 0;
+
+  const concentratedPoolLink =
+    osmosisQueries.queryCfmmToConcentratedLiquidityPoolLinks.get(
+      cfmmPoolId
+    ).concentratedLiquidityPoolId;
```

Comment:
Same as above

Reply:
Since we're using computed, MobX will return the same reference (and value, here, a string) as long as the query store remains the same